Break GitHub Deployment Action Into Jobs

dysbulic opened the pull request:
## Overview

**What features/fixes does this PR include?**

This PR takes what were a list of steps in the GitHub Action to deploy to Google Cloud Run and converts them into jobs with appropriate dependencies between them.

Because each job can potentially run on a separate instance, some of the steps like checking out the repository and authenticating to gcloud needed to be duplicated.

Also, this PR comments out the build step of the continuous integration test suite. A build is performed by the deployment, so that one was redundant.

Additionally, it moves the seeding step to before the frontend generation because the existing order was causing Next.js's initial cache to claim there are no players in the database.

**Please provide the GitHub issue number**

Closes #1062 & #1083.

## Follow up Improvement Ideas

Currently when the deployment is complete a comment is created in the PR giving the frontend and Hasura URLs. I would like to switch to using GitHub's deployments (#1080) which are more streamlined in the PR conversation flow & display under environments for the PR.

## Implementation

Environment variables are not shared between Action jobs. Also they can't be accessed from within the same step where they are defined. This PR defines an `env` job which generates all the various variables & exposes them as job outputs.

polimyl reviewed (changes requested):
the delete/remove job names are inconsistent and the database ones dont mention also deleting/creating an user.
otherwise lgtm

dysbulic commented:
@polimyl, I revamped the changes & I think I addressed your issues. The changes include:

* Removal of the commented out build portion of the CI Action
* Removal of references to the database being "local" in the seeding script
* Consistency in the naming of tasks for different subsystems
* Mention of the user creation task when setting up the db
* Division of the db and user creation into separate steps
* Addition of a step to remove the db user
* Migration from a comment in the PR to a GitHub deployment for notification to close #1080
* Division of long task commands into multiple lines

polimyl commented on .github/workflows/gcp-deploy.yaml:
```diff
@@ -19,14 +19,36 @@ env:
 
     steps:
-      - name: Cancel Existing Runs
+      - name: Cancel Running Actions
```
the previous "cancel existing runs" might actually be more accurate since this just cancels currently running runs of this action for this pr

polimyl commented on .github/workflows/gcp-deploy.yaml:
```diff
@@ -48,158 +70,192 @@ jobs:
 
     steps:
-      - name: Set Up gcloud CLI
+      - name: Set Up GCloud CLI
```
offical name is actually just "gcloud", all lowercase (according to its docs), same below

polimyl commented on .github/workflows/gcp-deploy.yaml:
```diff
@@ -48,158 +70,192 @@ jobs:
 
   create-db:
-    name: Create New Database
+    name: "Create New SQL User & Database"
```
why does this have quotation marks but the delete step doesnt? also the dleete step mentions hasura but this doesnt (fine with either, but both should be the same imo)

dysbulic replied:
The ampersand requires quotes.

polimyl replied:
the delete-db job also got an ampersand but no quotes though. guess you will have to add quotes there then

dysbulic replied:
I would've sworn that VSCode told me I needed quotes, but apparently not…

polimyl commented on .github/workflows/gcp-deploy.yaml:
```diff
@@ -48,158 +70,192 @@ jobs:
 
   delete-backend:
-    name: Remove Backend Image
+    name: Remove Backend Container Image
```
job id uses the word "delete" while the anme uses the word "remove"

polimyl commented on .github/workflows/gcp-deploy.yaml:
```diff
@@ -48,158 +70,192 @@ jobs:
           export_default_credentials: true
 
-      - name: Delete Backend Image
+      - name: "Delete Container: ${{needs.env.outputs.BACKEND_TAG}}"
```
i have recently learned the proper definitions of this terminology: a container image is a static file thats more or less just a snapshot of a container while a container is a running instance of a container image
you seem to mix those up in multiple places (i also did in the original workflow)

polimyl reviewed (changes requested):
might be a bit too nitpicky, but eh

polimyl reviewed (approved):
lgtm
